### PR TITLE
Throw error if no placements, so fallback will work

### PR DESCRIPTION
**Pull request by gcampo88**
#111

### CODE REVIEW

File: src/app/shared/dovetail/dovetail-audio.ts [hunk 1]
```diff
@@ -168,7 +168,9 @@ export class DovetailAudio extends ExtendableAudio {
     this._dovetailLoading = true;
     let promise = this.currentPromise = this.dovetailFetcher.fetch(url).then(
       result => {
-        if (this.currentPromise === promise) {
+        if (!result.request.placements.length) {
+          throw 'No placements in request; falling back to raw audio';
```
Comment (cavis) [inline src/app/shared/dovetail/dovetail-audio.ts]: Probably shouldn't throw a string in node - can cause problems in error handling.  I'd just `new Error()` that line.

**Comment by gcampo88** [comment]
back atcha @cavis

File: src/app/shared/dovetail/dovetail-audio.ts [hunk 2]
```diff
@@ -168,15 +168,18 @@ export class DovetailAudio extends ExtendableAudio {
...
+        } else if (this.currentPromise === promise) {
           let data: AllUnion  = [result, this.getArrangement(result.request)];
           return Promise.all<any>(data);
         }
       }
     ).then(([dovetailResponse, adzerkResponse]: AllResult) =>  {
       return this.calculateArrangement(dovetailResponse.arrangement, adzerkResponse);
     }).catch(error => {
-      return this.fallback(url);
+      let directUrl = this.dovetailFetcher.transform(url);
```
Comment (cavis) [inline src/app/shared/dovetail/dovetail-audio.ts]: Nitpick: there's an off chance that we mis-recognized a dovetail URL, and `dovetailFetcher.fetch()` got an actual mp3 file.  That would throw an error (cannot json-decode an mp3 file), and end up here.

So probably safer for this `.catch(` to continue redirecting to the original URL.

**Comment by gcampo88** [comment]
@cavis back to you! Went the customerror route

**Pull request merged**